docs/apache-airflow/extra-packages-ref.rst

```diff
@@ -97,7 +106,7 @@ with a consistent set of dependencies based on constraint files provided by Airf
 .. code-block:: bash
     :substitutions:
 
-    pip install apache-airflow[google,amazon,apache.spark]==|version| \
+    pip install apache-airflow[google,amazon,apache-spark]==|version| \
```

Comment:
Inconsistent or typo? Below you are consistent with `_`
```suggestion
    pip install apache-airflow[google,amazon,apache_spark]==|version| \
```

Reply:
I could not decide what form of extras to choose - I spoke to @uranusjr about it - but I decided to go full-in on using `-` following (BTW where @uranusjr himself was mentioned) https://peps.python.org/pep-0685/ 

While in some cases it generates warning and `pip` itself is not yet fully PEP-685 compliant https://github.com/pypa/pip/issues/11445 - I believe (and I would love @uranusjr to confirm that) - it's perfectly fine and backwards compatible to change all our extras to follow it and change them to `-` - as pretty much all modern-ish tools will correctly normalize any specification that contains `_` or `.` to one defined with `-`.

I will keep that open and wait for @uranusjr's coment on that. In case there would be any backwards-compatibility issues foreseen , we could still get all deps to use `-` and generate deprecated aliases in pyproject.toml - but I would rather avoid it, if we think it's `pretty much backwards compatible unless you user is using some very weird and old tooling`